### PR TITLE
Transition to ImgIx 

### PR DESCRIPTION
**Pull request by pldespaigne**
solve #3283

### CODE REVIEW

File: libs/media/src/lib/+state/media.firestore.ts [hunk 1]
```diff
@@ -38,47 +38,27 @@ export function createHostedMedia(media?: Partial<HostedMedia>): HostedMedia {
   }
 }
 
-
 /**
  * This array contains the different image sizes.
- * @note it **does not** contains the `original` key
  */
-export const imgSizeDirectory = ['lg', 'md', 'xs', 'fallback'] as const;
+export const imgSizeDirectory = ['lg', 'md', 'xs'] as const;
```
Comment (GrandSchtroumpf) [inline libs/media/src/lib/+state/media.firestore.ts:44]: If it's only used in the frontend can you move it into the directive direcly ?

File: libs/media/src/lib/+state/media.firestore.ts [hunk 2]
```diff
@@ -38,47 +38,27 @@ export function createHostedMedia(media?: Partial<HostedMedia>): HostedMedia {
...
-    return { xs: 50, md: 100, lg: 300, fallback: 1024 };
-  } else if (url.includes('logo')) {
-    return { xs: 50, md: 100, lg: 300, fallback: 1024 };
-  } else if (url.includes('poster')) {
-    return { xs: 200, md: 400, lg: 600, fallback: 1024 };
-  } else if (url.includes('banner')) {
-    return { xs: 300, md: 600, lg: 1200, fallback: 1024 };
-  } else if (url.includes('still')) {
-    return { xs: 50, md: 100, lg: 200, fallback: 1024 };
+export function getImgSize(ref: string) {
```
Comment (GrandSchtroumpf) [inline libs/media/src/lib/+state/media.firestore.ts:49]: Same as above.

File: apps/catalog/catalog/src/app/dashboard/movie-tunnel/chain-of-titles/chain-of-titles.component.html [hunk 3]
```diff
@@ -8,17 +8,13 @@ <h1 subtitle>Chain of Titles</h1>
...
       </mat-card-header>
       <mat-card-content>
-        <ng-container *ngFor="let control of chainOfTitles.controls; let i = index; let last = last">
+        <ng-container *ngFor="let control of chainOfTitles; let i = index; let last = last">
           <file-upload
             test-id="chain-titles"
-            path="pdf"
-            uploadOnFirestore="true"
-            return="imgRef"
+            storagePath=""
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/dashboard/movie-tunnel/chain-of-titles/chain-of-titles.component.html:14]: Could we make it optional with a default value in the component if we only give empty string ?

File: libs/media/src/lib/directives/image-reference/background-reference.directive.ts [hunk 4]
```diff
@@ -19,12 +19,13 @@ export class BackgroundReferenceDirective implements OnInit, OnDestroy {
...
   /** Set background-image attribute in any html tag with the url stored in firestore.
    *  If path is wrong, src will be set with provided placeholder or empty string */
-  @Input() set bgRef(image: ImgRef) {
+  @Input() set bgRef(image: HostedMedia) {
     if(!image){
       this.ref$.next('');
     } try {
-      const url = getImageUrl(image);
-      this.ref$.next(url);
+      // TODO IMPLEMENT IMGIX issue#3283
```
Comment (GrandSchtroumpf) [inline libs/media/src/lib/directives/image-reference/background-reference.directive.ts:26]: Shoudn't it be done in this PR ?

File: libs/media/src/lib/directives/image-reference/image-reference.directive.ts [hunk 5]
```diff
@@ -62,16 +78,44 @@ export class ImageReferenceDirective implements OnInit, OnDestroy {
...
+
+      this.cdr.markForCheck()
+    });
   }
 
   ngOnDestroy() {
     this.sub.unsubscribe();
   }
+
+  @HostListener('error')
```
Comment (GrandSchtroumpf) [inline libs/media/src/lib/directives/image-reference/image-reference.directive.ts:110]: Put the hostlistener just after the constructor

File: libs/media/src/lib/directives/image-reference/imgix-helpers.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+ * @example
+ * const param: ImageParameters = {
+ *   fit: 'crop',
+ *   width: 100,
+ *   height: 100
+ * };
+ * formatParameters(param); // '?fit=crop&w=100&h=100&'
+ */
+export function formatParameters(parameters: ImageParameters): string {
+  let query = '';
```
Comment (GrandSchtroumpf) [inline libs/media/src/lib/directives/image-reference/imgix-helpers.ts:26]: Personaly I would have done : 
```typescript
const query = Object.entries(paramters)
  .filter(([key, value]) => !!value)
  .map(([key, value]) => `${key}=${value}`)
  .join('&')
```
But then "width" becomes "w" & same for height.
But I think it's nice to have the same keys than imgix in the end.

File: libs/utils/src/lib/breakpoint/breakpoints.service.ts [hunk 7]
```diff
@@ -1,35 +1,35 @@
 import { Injectable } from "@angular/core";
 import { BreakpointObserver, Breakpoints } from "@angular/cdk/layout";
 import { shareReplay, distinctUntilChanged, map } from "rxjs/operators";
-import { Observable } from "rxjs";
+import { Observable, combineLatest } from "rxjs";
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/breakpoint/breakpoints.service.ts:4]: Looks like you don't need combineLatest anymore

File: libs/media/src/lib/+state/media.model.ts [hunk 8]
```diff
@@ -59,6 +59,18 @@ function mediaNeedsUpdate(media: HostedMediaFormValue) {
   return media.delete || (!!media.ref && !!media.blobOrFile);
 }
 
+export function getFileNameFromPath(path: string) {
+  return path.split('/').pop()
+}
+
+
+// TODO issue#3283 DO WE  REALLY NEED ALL THE CODE BELLOW ???
```
[the remark this answers is not in the document]
Comment (pldespaigne) [reply libs/media/src/lib/+state/media.model.ts:67]: I really like this kind of comment ! ^^

File: apps/catalog/catalog/src/app/dashboard/movie-tunnel/chain-of-titles/chain-of-titles.component.html (hunk 3 above)
Comment (pldespaigne) [reply apps/catalog/catalog/src/app/dashboard/movie-tunnel/chain-of-titles/chain-of-titles.component.html:14]: No, we need to know where we will upload the file when the user hit the save button, here it's a weird behavior.
We should either fix it or maybe it's the sign that we need another component.

File: libs/media/src/lib/directives/image-reference/background-reference.directive.ts (hunk 4 above)
Comment (pldespaigne) [reply libs/media/src/lib/directives/image-reference/background-reference.directive.ts:26]: indeed, but you are too fast ⚡

**Pull request merged**